src/BotPlutusInterface/PreBalance.hs

```diff
@@ -261,12 +262,13 @@ balanceNonAdaOuts ownPkh utxos tx =
 {- | Add the required signatorioes to the transaction. Be aware the the signature itself is invalid,
  and will be ignored. Only the pub key hashes are used, mapped to signing key files on disk.
 -}
-addSignatories :: PubKeyHash -> Map PubKeyHash PrivateKey -> [PubKeyHash] -> Tx -> Either Text Tx
+addSignatories :: PubKeyHash -> Map PubKeyHash DummyPrivKey -> [PubKeyHash] -> Tx -> Either Text Tx
 addSignatories ownPkh privKeys pkhs tx =
   foldM
     ( \tx' pkh ->
         case Map.lookup pkh privKeys of
-          Just privKey -> Right $ Tx.addSignature' privKey tx'
+          Just (FromSKey privKey) -> Right $ Tx.addSignature' privKey tx'
+          Just (FromVKey privKey) -> Right $ Tx.addSignature' privKey tx'
```

Comment:
Can't you use `unDummyPrivateKey` here over unpacking directly?

Reply:
Fixed.

Reply:
Fixed.